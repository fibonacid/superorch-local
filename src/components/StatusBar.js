import React, {Component} from 'react'
import {connect} from 'react-redux';
import styled from 'styled-components'

const StyledContainer = styled.div`
  padding: 5px;
  text-align: right;
  font-size: 14px;
`;

const StyledList = styled.div`
  /* */  
`;

class StatusBar extends Component {
  render() {
    return (
      <StyledContainer>
        {this.props.connected ?
          <p style={{color: "green"}}>connected</p> :
          <p style={{color: "red"}}>not connected</p>
        }
      </StyledContainer>
    )
  }
}

const mapStateToProps = state => ({
  connected: state.base.connected
});

export default connect(mapStateToProps)(StatusBar);
