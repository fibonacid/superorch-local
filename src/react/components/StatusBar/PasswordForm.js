import React, { useState, useCallback } from "react";
import styled from "styled-components";

const StyledSpan = styled.span`
  padding: 0 10px;
`;

function PasswordForm(props) {
  const [edit, setEdit] = useState(false);
  const [value, setValue] = useState("");

  const toggleEdit = useCallback(() => {
    setEdit(!edit);
  }, []);

  const handleInputChange = useCallback(event => {
    setValue(event.target.value);
  }, []);

  const handleDisable = useCallback(() => {
    props.disablePassword();
  }, []);

  const handleDone = useCallback(() => {
    setEdit(false);
    props.setPassword(value);
  }, [value]);

  const handleCancel = useCallback(() => {
    setEdit(false);
  }, []);

  const { password, requirePassword } = props.server;

  return (
    <div>
      {!edit ? (
        <>
          {requirePassword ? (
            <>
              <StyledSpan>{password}</StyledSpan>
              <button onClick={toggleEdit}>CHANGE</button>
              <button onClick={handleDisable}>DISABLE</button>
            </>
          ) : (
            <button onClick={toggleEdit}>SET PASSWORD</button>
          )}
        </>
      ) : (
        <>
          <input onChange={handleInputChange} value={value} />
          <button type="submit" onClick={handleDone}>
            DONE
          </button>
          <button onClick={handleCancel}>CANCEL</button>
        </>
      )}
    </div>
  );
}

export default PasswordForm;
