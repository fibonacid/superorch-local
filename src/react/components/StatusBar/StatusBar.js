import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components/macro";

const StyledContainer = styled.div`
  padding: 5px;
  font-size: 14px;
  flex: 0 0 auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const StyledConnected = styled.span`
  color: ${props => (props.isConnected ? "green" : "red")};
`;

function StatusBar(props) {
  const { appName, appVersion } = props;
  return (
    <StyledContainer data-test={"StatusBarComponent"}>
      <span data-test={"app-info"}>
        {appName} {appVersion}
      </span>
      <StyledConnected isConnected={props.isConnected}>
        {props.isConnected ? "connected" : "not connected"}
      </StyledConnected>
    </StyledContainer>
  );
}

StatusBar.propTypes = {
  appName: PropTypes.string,
  appVersion: PropTypes.string,
  isConnected: PropTypes.bool.isRequired
};

export default StatusBar;
