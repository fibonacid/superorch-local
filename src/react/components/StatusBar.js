import React from 'react'
import * as styled from './styles/StatusBar.styles';

function StatusBar(props) {
  const { appName, appVersion } = props;
  return (
    <styled.Container
      data-test={'StatusBarComponent'} >
      <span data-test={'app-info'}>
        {appName} {appVersion}
    </span>
    </styled.Container>
  )
}

export default StatusBar;
