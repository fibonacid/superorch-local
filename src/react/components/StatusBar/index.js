import { connect } from "react-redux";
import StatusBar from "./StatusBar";

const mapStateToProps = state => ({
  appName: state.app.name,
  appVersion: state.app.version,
  isConnected: state.websocket.isConnected
});

export default connect(mapStateToProps, null)(StatusBar);
