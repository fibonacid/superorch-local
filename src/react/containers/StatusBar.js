import {connect} from "react-redux";
import StatusBar from "../components/StatusBar";

const mapStateToProps = state => ({
  appName: state.app.name,
  appVersion: state.app.version
});

export default connect(mapStateToProps, null)(StatusBar);
