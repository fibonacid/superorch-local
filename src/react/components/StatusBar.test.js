import React from "react";
import {shallow} from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from "../store";
import StatusBar from "./StatusBar";

const mockStore = configureStore();

describe("StatusBar", () => {
  it ('should render my component', () => {
    // mount component (shallow)
    const component = shallow(
      <Provider store={mockStore}>
        <StatusBar/>
      </Provider>
    );
    // unmount component
    component.unmount();
  });
});
