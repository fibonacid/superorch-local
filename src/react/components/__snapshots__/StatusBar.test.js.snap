// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar Component When has props Should renderer without errors 1`] = `
<StatusBarstyles__Container
  data-test="StatusBarComponent"
>
  <span
    data-test="app-info"
  >
    testApp
     
    0.0.0.test
  </span>
</StatusBarstyles__Container>
`;
